department-of-veterans-affairs/caseflow
# APPEALS-6061 Add a check box on DAS

Resolves [APPEALS-6061](https://vajira.max.gov/browse/APPEALS-6061)

### Description
As a Caseflow user and judge, I need the ability to determine and document if the drafted decision by an attorney was provided on a timely or untimely manner.  This new evaluation criteria needs to be provided on the DAS where the drafted decision is currently evaluated.   


### Acceptance Criteria
- [ ] On the Evaluate Decision page expand the section called ‘Case Timeliness’
- [ ] Add a radio button, on the ‘Case Timeliness’ section, that has:
  - [ ] A label that reads: ‘Was this decision timely?’
  - [ ] Selectable options: ‘Timely’ and ‘Untimely’
- [ ] New timely radio button is a mandatory field
- [ ] When the Evaluate Decision page is initially displayed, the radio buttons are empty (unmarked)
- [ ] Value stored from the radio buttons for ‘Timely’ and ‘Untimely’ is available for future Tableau reporting and stored in the same table (judge_case_reviews) where other evaluation fields from the Evaluate Decision page are stored.
- [ ]  Radio buttons are tabbed to after previous interactive element on the page
- [ ] If the user presses the ‘Back’ command, at the bottom right corner of the page, the values entered in the ‘Timely’ radio button is cleared.  If user navigates back to the Evaluate Decision page, the value is empty. (See Acceptance Criteria 4)
- [ ] If the user presses the ‘Cancel’ command, at the bottom left corner of the page:
  - [ ] The existing Cancel confirmation modal is displayed
  - [ ] The values entered in the ‘Timely’ radio button is cleared.  If user navigates back to the Evaluate Decision page, the value is empty. (See Acceptance Criteria 4)
- [ ] If the user presses the ‘Continue’ command, at the bottom right corner of the page, and the Timely radio button is empty:
  - [ ] Page will not allow the user to continue forward 
  - [ ] The section will be highlighted by a red line on the left of the options
  - [ ] A label that reads: ‘Choose one’ is displayed in red above the radio buttons
- [ ] Appeals that with decisions drafted and evaluated before the implementation of this new evaluation field, will have this value as NULL.  Old cases will not be updated to reflect a value on this new evaluation field.


### Testing Plan
1. Run `make db-migrate`
2. Ensure feature toggle `das_case_timeliness` is enabled
3. Switch to a Judge User (e.g. BVAAABSHIRE) and navigate to their queue.
4. Locate an appeal that has a "Review" Task and click on the link to go to the Case Details Page.
5. In the Actions dropdown, select "Ready for Dispatch"
6. Select 'no special issues' and continue filling out the forms
7. When you get to the Evaluate Decision page, ensure that the "Was this decision timely?" Question and radio options appear in the Case Timeliness section.
8. Fill out other required fields 
9. Click Continue
10. Run `make c`
11. Run command `ActiveRecord::Base.connection.execute("select * from judge_case_reviews where timeliness is not null").values`
12. Observe that a value of timely or untimely was stored for above review


### User Facing Changes

 BEFORE
![image](https://user-images.githubusercontent.com/1094999/180833158-84b96250-d49a-49d2-8378-c9f2c729a1ec.png)
AFTER
![image](https://user-images.githubusercontent.com/1094999/180833193-fa161d00-6db2-44e2-8c4d-71320d994a55.png)
 ---|---

### Code Documentation Updates

### Database Changes
*Only for Schema Changes*

Adds the column timeliness to the table judge_case_reviews
